modules/javafx.controls/src/main/java/javafx/scene/control/skin/MenuBarSkin.java

```diff
@@ -229,17 +229,36 @@ public MenuBarSkin(final MenuBar control) {
 
         ListenerHelper lh = ListenerHelper.get(this);
 
+        if (Platform.isFxApplicationThread()) {
+            if (Toolkit.getToolkit().getSystemMenu().isSupported()) {
```

Comment:
You could move this check to the outer scope, because if it evalutes to false, we can skip both branches of `if (Platform.isFxApplicationThread()) {` completely.

Reply:
not sure what you mean exactly.
Toolkit.getToolkit().getSystemMenu().isSupported() must be called in the fx application thread, and the rest of the changes were done to minimize the structural changes.